app/src/main/java/com/asfoundation/wallet/promotions/voucher/EVoucherDetailsInteractor.kt

```diff
@@ -0,0 +1,21 @@
+package com.asfoundation.wallet.promotions.voucher
+
+import java.util.*
+
+class EVoucherDetailsInteractor {
+
+  fun getDiamondModels(): List<DiamondsButtonModel> {
```

Comment:
I'd suggest this returning a Single instead and setting up its chain in the presenter side, since the WS will return a single anyway.